sw/device/lib/dif/README.md

```diff
@@ -12,3 +12,56 @@ libraries.
 There is one DIF library per hardware IP, and each one contains the DIFs
 required to actuate all of the specification-required functionality of the
 hardware they are written for.
+
+## DIF Style Guide
+
+DIFs are very low-level software, so they have a more rigorous coding style than
+other parts of the codebase.
+
+DIFs should follow the [OpenTitan C/C++ style
+guide](https://docs.opentitan.org/doc/rm/c_cpp_coding_style/) where it does not
+contradict with the guidelines below.
+
+The guidelines below apply to writing DIFs, and code should be written in a
+similar style to the existing DIF libraries in this directory.
+
+### DIF Library Guidance
+
+* DIF libraries must be written in C.
+* DIF libraries can only depend on the headers and libraries in the
+  `sw/device/lib/base` directory.
+* DIF libraries must not depend on other DIF libraries. Exercising DIF
+  functionality may require an environment set up using another DIF library, but
+  DIFs must not call DIFs in other DIF libraries.
+
+### DIF Guidance
+
+* DIF declarations must match their definitions exactly.
+  * Scalar arguments should not be marked `const` in DIF signatures.
+
+* DIFs must use enum return codes rather than booleans for reporting errors. If
+  a DIF can either error or produce a value, it should return an enum return
+  code, and use an out-parameter for returning the produced value.
+
+* DIFs must check their arguments against preconditions using "guard
+  statements". A guard statement is a simple if statement at the start of a
+  function which only returns an error code if the preconditions are not met.
+  Guard statements must cover the following checks:
+  * DIFs must ensure their pointer arguments are non-null, unless that pointer
```

Comment:
I think this should clarify that `mmio_region_t` is not a pointer parameter that can be meaningfully checked for nullness. DIFs should never ever access the pointer inside, even for a null check, since that pointer even existing is an implementation detail (which tests, for example, may violate).

Reply:
`mmio_region_t` is not a pointer parameter. It is a struct parameter. Yes, I will include that you should not be looking inside a `mmio_region_t`.